### PR TITLE
A step by step example of how to make thing data driven.

### PR DESCRIPTION
**Pull request by gigamonkey**
You should look at the changes in this PR commit by commit because I made the changes in very small steps to show exactly how I would do something like this. I suspect if you do the same thing to the rotation functions for other shapes you'll probably end up being able turn the `rotateJ` function I added into just `rotate` and have it be completely generic. I also suspect that there may be other mathematical regularities that we can tease out but it will be easier to see them with a bit more tidying.

### CODE REVIEW

**Comment by gigamonkey** [comment]
BTW, I made this as a PR against the branch of your other PR. So if you merge this PR it will go into the branch of your other PR which you can then merge into `main`. You should thus merge this one first. (This is where working with git can get a bit confusing but if we make a mess of things I can help sort it out.)

File: script.js [hunk 1]
```diff
@@ -411,6 +411,15 @@ function rotateJC4JCC3(){
   curBlock[3].yPos += -20
 }
 
+function rotateJ(values) {
+  curBlock[0].xPos += vaues[0].x;
```
Comment (gigamonkey) [inline script.js]: Whoops, typo on this line in the variable name. Fixed later.

File: script.js [hunk 2]
```diff
@@ -396,38 +396,20 @@ function rotateJ(values) {
   }
 }
 
-function checkJRotationClockwise(){
+function checkJRotation(clockwise){
   if (curBlock[0].xPos <= curBlock[3].xPos && curBlock[0].yPos <= curBlock[3].yPos) {
-    return 0;
+    return clockwise ? 0 : 2;
```
Comment (gigamonkey) [inline script.js:401]: This thing with the ? and : is an operator you may not have seen before called the ternary operator. It's kind of like an if except its an expression that produces a value so this line will return 0 if clockwise is true and 2 if it's false.

**Comment by NicholasRebusi** [comment]
I beleive I understand what your doing here. I'll try to implement this with the other blocks in the coming days.  Thanks for the help.

**Review by NicholasRebusi — APPROVED** [review]
Looks good and I think I understand it. Thanks for the help.

**Pull request merged**